File: Arrays/TwoSum.js
var twoSum = function(nums, target) {
    for (let firstIndex = 0; firstIndex < nums.length - 1; firstIndex++) {
        for (let secondIndex = firstIndex + 1; secondIndex < nums.length; secondIndex++) {
            if (nums[firstIndex] + nums[secondIndex] === target) {
                return [firstIndex, secondIndex];
            }
        }
    }


    return [];
};

File: Arrays/TwoSum.c
#include <stdio.h>
#include <stdlib.h>


int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    static int resultIndices[2];


    *returnSize = 2;


    for (int firstIndex = 0; firstIndex < numsSize - 1; firstIndex++) {
        for (int secondIndex = firstIndex + 1; secondIndex < numsSize; secondIndex++) {
            if (nums[firstIndex] + nums[secondIndex] == target) {
                resultIndices[0] = firstIndex;
                resultIndices[1] = secondIndex;


                return resultIndices;           
            }
        }
    }


    return NULL;
}
